File: teceze/teceze/report/holiday_details/holiday_details.js
// Copyright (c) 2026, Teceze Consultancy Pvt. Ltd. and contributors
// For license information, please see license.txt

frappe.query_reports["Holiday Details"] = {
    filters: [
        {
            fieldname: "holiday_name",
            label: __("Holiday List"),
            fieldtype: "Link",
            options: "Holiday List",
            reqd: 1,
			default : "India-2026-2027"
        }
    ],

    formatter: function (value, row, column, data, default_formatter) {
        value = default_formatter(value, row, column, data);

        if (column.fieldname === "holiday_date") {
            return `<span style="
                background-color: #fff3cd;
                color: #856404;
                padding: 4px 8px;
                border-radius: 4px;
                font-weight: 600;
            ">${value}</span>`;
        }

        if (column.fieldname === "description") {
            return `<span style="
                color: #1565c0;
                font-weight: 600;
            ">${value}</span>`;
        }

        if (column.fieldname === "custom_status") {
            return `<span style="
                color: #6a1b9a;
                font-weight: 600;
            ">${value}</span>`;
        }

        return value;
    }
};

File: teceze/teceze/report/holiday_details/holiday_details.py
# Copyright (c) 2026, Teceze Consultancy Pvt. Ltd. and contributors
# For license information, please see license.txt

import frappe


def execute(filters=None):
    filters = filters or {}

    columns = [
        {
            "label": "Holiday List",
            "fieldname": "holiday_list",
            "fieldtype": "Link",
            "options": "Holiday List",
            "width": 220,
			"hidden": 1
        },
        {
            "label": "Holiday Date",
            "fieldname": "holiday_date",
            "fieldtype": "Date",
            "width": 130,
        },
        {
            "label": "Holiday Name",
            "fieldname": "description",
            "fieldtype": "Data",
            "width": 300,
        },
		 {
            "label": "Status",
            "fieldname": "custom_status",
            "fieldtype": "Data",
            "width": 300,
        },
    ]

    conditions = [
        "hd.weekly_off = 0"
    ]

    values = {}

    if filters.get("holiday_name"):
        conditions.append("h.name = %(holiday_name)s")
        values["holiday_name"] = filters["holiday_name"]

    data = frappe.db.sql(
        f"""
        SELECT
            h.name AS holiday_list,
            hd.holiday_date,
            hd.description,
			hd.custom_status
        FROM `tabHoliday List` h
        INNER JOIN `tabHoliday` hd
            ON hd.parent = h.name
        WHERE {" AND ".join(conditions)}
        ORDER BY hd.holiday_date
        """,
        values,
        as_dict=True,
    )

    return columns, data